x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/DefaultCheckpointProvider.java

```diff
@@ -93,7 +94,7 @@ public void createNextCheckpoint(final TransformCheckpoint lastCheckpoint, final
         final long timestamp = clock.millis();
         final long checkpoint = TransformCheckpoint.isNullOrEmpty(lastCheckpoint) ? 1 : lastCheckpoint.getCheckpoint() + 1;
 
-        getIndexCheckpoints(ActionListener.wrap(checkpointsByIndex -> {
+        getIndexCheckpoints(null, ActionListener.wrap(checkpointsByIndex -> {
```

Comment:
It might be better than nothing to supply a timeout here, even if it's huge, say 12 hours.

It might help to release cluster resources earlier if someone configures a transform that ends up checkpointing 100000 searchable snapshot indices that all have to be retrieved from blob storage.